zerver/lib/markdown/__init__.py

```diff
@@ -956,9 +956,15 @@ def set_text(text: str) -> None:
             Helper to set the text or the tail of the current_node
             """
             if current_node == p:
-                current_node.text = text
+                if current_node.text is None:
+                    current_node.text = text
+                else:
+                    current_node.text += text
             else:
-                current_node.tail = text
+                if current_node.tail is None:
+                    current_node.tail = text
+                else:
+                    current_node.tail += text
```

Comment:
I think it's a little confusing to change this function's semantics that way -- would it be clearer to just have an `append_text` function that does the append version?  I certainly would read `set_text` as an overwrite, not an append, if something was already there.

Reply:
Okay, but that would probably make look the code messy, as each time, before making a call to any of those two functions, I need to check if `current_node.text` is None or not.
Probably changing the name of the function (and description to what it does), `set_text` to `amend_text` or `add_text` would do the job ?

Reply:
Yeah, or  `set_or_append` or `set_text(append_if_present=True)` could be OK -- just anything to make clear what the actual behavior is.